api/types/header/convert/legacy/header.go

```diff
@@ -24,9 +24,11 @@ import (
 // FromHeaderMetadata will convert a *header.Metadata object to this metadata object.
 // TODO: Remove this once we get rid of the old Metadata object.
 func FromHeaderMetadata(metadata header.Metadata) types.Metadata {
+	expires := metadata.Expires
 	return types.Metadata{
 		ID:          metadata.ID,
 		Name:        metadata.Name,
+		Expires:     &expires,
```

Comment:
This won't work?
```suggestion
		Expires:     &metadata.Expires,
```

Reply:
It will, but I was worried about keeping the pointer for whatever reason. I can change it, however.